### PR TITLE
curl: update to 8.7.1

### PR DESCRIPTION
**Pull request by stangri**
Maintainer: me
Compile tested: x86_64, Sophos XG-135r3, OpenWrt 23.05.2
Run tested: x86_64, Sophos XG-135r3, OpenWrt 23.05.2

Description:
* update to 8.7.1: https://curl.se/changes.html#8_7_1
* use the new --disable-docs flag for configure
* update 200-no_docs_tests.patch
* switch to APK-compatible revision

### CODE REVIEW

**Comment by liudongmiao** [comment]
@stangri As curl 8.7.1 (actually since 8.6.0) support http/3 with openssl 3.2+.
Could you help to support http/3 in curl with an update for openssl?

**Comment by liudongmiao** [comment]
#19382 With the openssl 3.2, support http/3 should be very easy (however curl mark http/3 with openssl 3.2 is experimental): https://curl.se/docs/http3.html

- ngtcp2 variant: quicltls (openssl with quic patch), nghttp3, ngtcp2 
- openssl variant: openssl to 3.2.1+

**Comment by liudongmiao** [comment]
@stangri And, is there any chance to switch from configure to cmake for curl?
cmake should be much more easier to support dynamic `CONFIGURE_ARGS`.

File: net/curl/Makefile [hunk 1]
```diff
@@ -9,15 +9,15 @@ include $(TOPDIR)/rules.mk
 include $(INCLUDE_DIR)/nls.mk
 
 PKG_NAME:=curl
-PKG_VERSION:=8.6.0
-PKG_RELEASE:=1
+PKG_VERSION:=8.7.1
+PKG_RELEASE:=r1
```
Comment (liudongmiao) [inline net/curl/Makefile:13]: why use `r1`?

Comment (1715173329) [reply net/curl/Makefile:13]: `r` will be appended automatically, do not add it here

Comment (stangri) [reply net/curl/Makefile:13]: @1715173329 For multitude of reasons, not a big fan of having different version numbers in the makefile and with the binary, so the build system will need to change to not prepend `r` if it already exists in the revision.

**Comment by stangri** [comment]
> @stangri And, is there any chance to switch from configure to cmake for curl? cmake should be much more easier to support dynamic `CONFIGURE_ARGS`.

You're welcome to submit a PR for the switch.

File: net/curl/Makefile (hunk 1 above)
Comment (1715173329) [reply net/curl/Makefile:13]: oh i see it, thanks for clarification!

**Pull request merged**